### ISSUE TITLE
Loop between Reset-GitHubConfiguration and Clear-GitHubAuthentication

### ISSUE THREAD

**Issue by smaglio81** [microsoft/PowerShellForGitHub#124]
## Description

The commands Reset-GitHubConfiguration and Clear-GitHubAuthentication can get into an infinite suggestion loop.

![image](https://user-images.githubusercontent.com/169953/61767158-280bb180-ad98-11e9-9cdc-07aa10a46bc5.png)

## What I was doing / Steps to Reproduce

I downloaded the code for the first time and I was attempting to create a new unit test. Running the new unit test in VSCode presented a series of informative messages and I went about trying to accomplish the correct setup.

I had set the `Tests/Config/Settings.ps1` file with my information, and as you can see from the screen shot, I had just entered my GitHub username/password into a PSCredential.

I then used `Set-GitHubAuthentication` to set the credential and attempted to rerun the unit test. I was running the new unit tests file individually, by opening it in VSCode and hitting `F5`. It resulted in the error from the screenshot. Please ignore that error, as my custom made / new code is not yet working.

However, it did result in an informative warning message.

I read (or misread) the warning message to indicate that I should run `Clear-GitHubAuthentication` to prevent the warning message from appearing in the future. (Even though I feel like that name means I'm clearing my authentication tokens).

After the `Clear` command ran, it left another warning message that I should run `Reset-GitHubConfiguration`.

Unfortunately, running the `Reset` command left a warning that I should run `Clear-GitHubAuthentication`.

Soo .... I figure that's not supposed create a warning message loop like that.

**Comment by HowardWolosky** [comment]
Thanks for the report, @smaglio81.  In this case, there isn't actually a loop.  I think the problem is the similarly worded warnings that now appear easy to misread.  I'm open to suggestions on how to reword them.

There are two things being manipulated here:  Authentication information and Setting configuration.

`Clear-GitHubAuthentication` leaves your settings configuration alone and _only_ clears out the authentication information that was cached.  When run, it provides the warning that the configuration settings were _not_ cleared (in the event that people had expected that they would be), and informs you that you can clear your configuration settings by calling `Reset-GitHubConfiguration`.

Meanwhile, `Reset-GitHubConfiguration` will reset your configuration settings, but will leave your authentication info alone.  It spits out that warning to make sure you understand that.

So...that means to clear out _both_ your cached authentication info _and_ your configuration settings, you need to call `Clear-GitHubAuthentication` _and_ `Reset-GitHubConfiguration`.

The loop doesn't actually exist.  It exists only because you kept running the commands unnecessarily.  But I don't want to just point the blame at user error.  If you have suggestions on how that could all be re-worded in order to avoid that misunderstanding from happening with anyone else in the future, I'm all ears.  Just re-activate this Issue with your suggestion.

Thanks again for the report.

**Comment by smaglio81** [comment]
Hmmm ... I think, for me, I would prefer to not have the warning messages. I felt like I understood what the command was doing from the name. The warning message made me think that I _needed_ to go run the other command.

But, I think what started it all was the warning message when running the unit tests.

So ... I have setup authentication using `Set-GitHubAuthentication` and I had updated the `\Tests\Config\Settings.ps1` with my information. Why do I continue to receive the warning message?

Looking into `Tests\Common.ps1`, it looks like that message is tied to the variable `$env:avAccessToken`  (appVeyorAccessToken). Should that message be tied to that variable? Or, should it be tied to `$script:accessTokenCredential`?

Also within `Tests\Common.ps1`, it looks like `Reset-GitHubConfiguration` is being called. This generated the original warning message that lead me to run the `Clear-GitHubAuthentication` command. Maybe if that warning message no longer appeared, it wouldn't become entangled with the other message when running the unit test.

Anyways. Thanks for the help! And the great module! I really love what's been done with the telemetry.

**Comment by HowardWolosky** [comment]
Thanks @smaglio81.  I'll reopen this to encourage continued discussion.

I'm open to a better check than this one, but I'm not convinced with yours.
https://github.com/microsoft/PowerShellForGitHub/blob/d6c3b5354bfaa6538123798cc97f684f273140de/Tests/Common.ps1#L39-L48

It's not _just_ that the authentication has been configured.  It's _also_ that a user has modified `Tests\Config\Settings.ps1`, because even if you configure _authentication_, you still don't have access to the `PowerShellForGitHubTeam` org to run the tests against _that_ account.

https://github.com/microsoft/PowerShellForGitHub/blob/d6c3b5354bfaa6538123798cc97f684f273140de/Tests/Config/Settings.ps1#L4-L8

To avoid seeing that big warning message, we'd have to know that the user _changed_ those values (unless they're me) _and_ have cached authentication credentials.

As for the warnings:
https://github.com/microsoft/PowerShellForGitHub/blob/d6c3b5354bfaa6538123798cc97f684f273140de/GitHubConfiguration.ps1#L507

https://github.com/microsoft/PowerShellForGitHub/blob/d6c3b5354bfaa6538123798cc97f684f273140de/GitHubConfiguration.ps1#L940

As I said, I'm open to changing those messages from `Warning` to `Verbose` and then adding a `.NOTE` section to each of those methods that mentions the same thing.

If you're interested in making those changes, I'd be happy to review and merge them.

**Comment by smaglio81** [comment]
Yeah, that makes sense. What if we use Get-FileHash to check if the file has been updated?

Here's a code example (my apologizes for not linking to a github repo; I'm nervous about updating my repo before the previous pull request has been finished):

https://gist.github.com/smaglio81/9bf69669740ae7e7126ca6111c6e5580

I've also got the `Verbose` and `.NOTES` changes in `GitHubConfiguration.ps1`, once we're ready.
